src/CoreLib/DSeq.fs

```diff
@@ -532,71 +537,88 @@ and [<AllowNullLiteral>]
     /// Return: 
     ///     false: blocking, need to resend the command at a later time
     ///     true: command processed. 
-    member internal x.ProcessJobCommand( queuePeer:NetworkCommandQueue, cmd:ControllerCommand, ms:StreamBase<byte> ) = 
-        let peeri = x.Cluster.SearchForEndPoint( queuePeer )
-        Logger.LogF( LogLevel.WildVerbose, ( fun _ -> sprintf "ProcessJobCommand, Map %A to peer %d" queuePeer.RemoteEndPoint peeri ))
-        try
+    member internal x.ProcessJobCommand( jobAction:SingleJobActionContainer, queuePeer:NetworkCommandQueue, cmd:ControllerCommand, ms:StreamBase<byte>, jobID: Guid ) = 
+//        using ( SingleJobActionContainer.TryFind(jobID)) ( fun jobAction -> 
+//            if Utils.IsNull jobAction then 
+//                Logger.LogF( jobID, LogLevel.Info, ( fun _ -> sprintf "[May be OK, Job cancelled] DStream.Callback, received command %A of payload %dB, but can't find job lifecycle object" 
+//                                                                        cmd ms.Length )    )
+//                true
+//            else
+//                try
+            let peeri = x.Cluster.SearchForEndPoint( queuePeer )
+            Logger.LogF( jobID, LogLevel.WildVerbose, ( fun _ -> sprintf "ProcessJobCommand, Map %A to peer %d" queuePeer.RemoteEndPoint peeri ))
             // Outgoing queue for the reply 
             match ( cmd.Verb, cmd.Noun ) with 
             | ( ControllerVerb.Acknowledge, ControllerNoun.DStream ) ->
                 true
             | ( ControllerVerb.SyncWrite, ControllerNoun.DStream ) ->
                 let meta = BlobMetadata.Unpack( ms )
-                Logger.LogF( LogLevel.WildVerbose, ( fun _ -> sprintf "Rcvd %A command %A from peer %d with %dB" (meta.ToString()) cmd peeri (ms.Length-ms.Position) ))
+                Logger.LogF( jobID, LogLevel.MildVerbose, ( fun _ -> sprintf "Rcvd %A command %A from peer %d with %dB" (meta.ToString()) cmd peeri (ms.Length-ms.Position) ))
                 let bRet = x.SyncReceiveFromPeer meta ms peeri 
 //                let ev = new ManualResetEvent(false)
 //                ev.Reset() |> ignore
 //                ev.WaitOne() |> ignore
                 if bRet && (DateTime.UtcNow - queuePeer.LastSendTicks).TotalMilliseconds > 5000. then 
 //                if bRet then 
-                    let msEcho = new MemStream( 128 )
+                    use msEcho = new MemStream( 128 )
```

Comment:
An estimation of the size of the message. With the change Sanjeev will introduce, I assume that all new MemStream of a size small enough will be equivalent.